rust/core-lib/src/plugins/catalog.rs

```diff
@@ -79,10 +84,24 @@ impl<'a> PluginCatalog {
         self.items.keys()
     }
 
+    /// Returns the plugin located at the provided file path.
+    #[allow(clippy::map_clone)]
+    pub fn get_from_path(&self, path: &PathBuf) -> Option<Arc<PluginDescription>> {
+        self.items
+            .values()
+            .find(|&v| v.exec_path.to_str().unwrap().contains(path.to_str().unwrap()))
+            .map(|plugin| plugin.clone())
```

Comment:
Maybe there is a more elegant way to do this since clippy was complaining?

Reply:
From the [Great List of Clippy Lints](https://rust-lang.github.io/rust-clippy/master/index.html): Checks for usage of iterator.map(|x| x.clone()) and suggests iterator.cloned() instead.

However here we aren't dealing with an Iterator anymore, since `find()` only returns a n `Option<T>`, in this case `Option<&Arc<PluginDescription>>` because `values()` always yields references. Seems like replacing the `map()` just `.cloned()` would do the trick?

Reply:
Also that clippy annotation can then be removed.

Reply:
Thanks, fixed it :)